pkg/reader/price_reader.go

```diff
@@ -66,23 +72,28 @@ type LatestRoundData struct {
 	AnsweredInRound *big.Int
 }
 
-func (pr *OnchainTokenPricesReader) GetFeeQuoterTokenUpdates(
+func (pr *priceReader) GetFeeQuoterTokenUpdates(
 	ctx context.Context,
 	tokens []ocr2types.Account,
+	chain ccipocr3.ChainSelector,
 ) (map[ocr2types.Account]plugintypes.TimestampedBig, error) {
-	if !pr.feeQuoterEnabled {
-		return make(map[ocr2types.Account]plugintypes.TimestampedBig), nil
+	updates := make([]plugintypes.TimestampedBig, len(tokens))
+	updateMap := make(map[ocr2types.Account]plugintypes.TimestampedBig)
+
+	feeQuoterAddress, err := pr.ccipReader.GetContractAddress(consts.ContractNameFeeQuoter, chain)
+	if err != nil {
+		pr.lggr.Debugf("failed to get fee quoter address for chain %d: %v", chain, err)
+		return updateMap, nil
 	}
-	var updates []plugintypes.TimestampedBig
 
 	boundContract := commontypes.BoundContract{
-		Address: string(pr.FeeQuoterAddress),
+		Address: string(feeQuoterAddress),
```

Comment:
How does this work in the tests? This seems wrong

Reply:
Why does it seem wrong? Did you see the code just above it that sets the feeQuoter?

Reply:
`string([]byte)` doesn't give you a hex address, but base64 I think. I'm pretty sure the chain reader needs a hex address. So how does this work 😄

Reply:
Good catch man. Seems it's failing gracefully and that's why the test is passing